drivers/sensor/xbr818/xbr818.c

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright (c) 2024 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT phosense_xbr818
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(XBR818, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "xbr818.h"
+
+static int xbr818_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct xbr818_config *config = dev->config;
+	struct xbr818_data *data = dev->data;
+	int ret;
+
+	if (chan != SENSOR_CHAN_PROX && chan != SENSOR_CHAN_ALL) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_get_dt(&config->io_val);
+	if (ret < 0) {
+		return ret;
+	}
+	data->value = (ret == 1 ? true : false);
+
+	return 0;
+}
+
+static int xbr818_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct xbr818_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_PROX) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	val->val1 = (data->value ? 1 : 0);
+	val->val2 = 0;
+
+	return 0;
+}
+
+static int xbr818_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, const struct sensor_value *val)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+	uint32_t tmp;
+	uint64_t tmpf;
+
+	if (chan != SENSOR_CHAN_PROX) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	if (val->val1 < 0) {
+		return -EINVAL;
+	}
+
+	gpio_pin_set_dt(&config->i2c_en, 1);
```

Comment:
Handle return value [gpio_pin_set_dt](https://docs.zephyrproject.org/latest/doxygen/html/group__gpio__interface.html#ga541064fb9e575c0c559c101754466fa8)
```suggestion
	(void)gpio_pin_set_dt(&config->i2c_en, 1);
```
or
```suggestion
	ret = gpio_pin_set_dt(&config->i2c_en, 1);
	if (ret) {
	    return ret;
	}
```

Reply:
Very good point, done.